python/cudf/cudf/core/column/datetime.py

```diff
@@ -210,13 +210,6 @@ def fillna(self, fill_value):
             fill_value = column.as_column(fill_value, nan_as_null=False)
 
         result = libcudf.replace.replace_nulls(self, fill_value)
-        result = column.build_column(
-            result.base_data,
-            result.dtype,
-            mask=None,
-            offset=result.offset,
-            size=result.size,
-        )
```

Comment:
IIRC `replace_nulls` used to not get rid of the mask. Does it now?

Reply:
It is not getting rid of mask, but if the `fill_value` has `<NA>`/`null` values, that is applied onto `self` and the `result` will have a new mask based on the locations of null values in `self` and `fill_value`